virttest/utils_test/libvirt.py

```diff
@@ -1151,22 +1151,36 @@ def thread_func_migration(self, vm, desturi, options=None,
                 self.RET_LOCK.release()
 
     def do_migration(self, vms, srcuri, desturi, migration_type, options=None,
-                     thread_timeout=60, ignore_status=False):
+                     thread_timeout=60, ignore_status=False, func=None,
+                     func_params=None):
```

Comment:
It's okay to add optional parameters here. But you should consider to use kwargs to avoid the parameter numbers become more and more.